### PR TITLE
Re-enable ocaml_intrinsics on macos

### PR DESCRIPTION
**Pull request by gretay-js**
I'd like to re-enable `ocaml_intrinsics`  after the fix in https://github.com/janestreet/ocaml_intrinsics/pull/5 is merged.

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
As far as i understand it it requires a new release. Modifying the lastest release would not work

File: packages/ocaml_intrinsics/ocaml_intrinsics.v0.15.0/opam [hunk 1]
```diff
@@ -14,7 +14,7 @@ depends: [
   "dune"              {>= "2.0.0"}
   "dune-configurator"
 ]
-available: (arch = "x86_32" | arch = "x86_64") & os != "win32" & os != "macos"
+available: (arch = "x86_32" | arch = "x86_64") & os != "win32"
```
Comment (kit-ty-kate) [inline packages/ocaml_intrinsics/ocaml_intrinsics.v0.15.0/opam:17]: this also constraints newer macs out (arm64)

**Comment by kit-ty-kate** [comment]
For the record `ocaml_intrinsics` itself "compiles" fine on macOS/arm64 but subsequent packages using it would fail. For example:
```
#=== ERROR while compiling core_unix.v0.15.0 ==================================#
# context     2.2.0~alpha~dev | macos/arm64 | ocaml-base-compiler.4.14.0 | git+https://github.com/ocaml/opam-repository.git
# path        ~/.opam/default/.opam-switch/build/core_unix.v0.15.0
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p core_unix -j 9
# exit-code   1
# env-file    ~/.opam/log/core_unix-99966-49e66d.env
# output-file ~/.opam/log/core_unix-99966-49e66d.out
### output ###
# [...]
# /Users/kit_ty_kate/.opam/default/lib/ocaml/caml/misc.h:57:3: note: expanded from macro 'CAML_DEPRECATED'
#   CAML_PREPROWARNING(name1 is deprecated: use name2 instead)
#   ^
# /Users/kit_ty_kate/.opam/default/lib/ocaml/caml/misc.h:55:31: note: expanded from macro 'CAML_PREPROWARNING'
# #define CAML_PREPROWARNING(x) _Pragma(CAML_MAKEWARNING2(x))
#                               ^
# <scratch space>:59:6: note: expanded from here
#  GCC warning "\"invalid_argument\" is deprecated: use \"caml_invalid_argument\" instead"
#      ^
# 4 warnings generated.
# (cd _build/default/unix_pseudo_terminal/src && /bin/bash -e -u -o pipefail -c './discover.sh config_ext.h /Users/kit_ty_kate/.opam/default/lib/jst-config/config.h')
# getconf: no such configuration parameter `GNU_LIBC_VERSION'
```
```
(cd _build/default/filename_unix/src && /usr/bin/cc -O2 -fno-strict-aliasing -fwrapv -pthread -D_FILE_OFFSET_BITS=64 -O2 -fno-strict-aliasing -fwrapv -pthread -D_LARGEFILE64_SOURCE -g -I /Users/kit_ty_kate/.opam/default/lib/ocaml -I /Users/kit_ty_kate/.opam/default/lib/base -I /Users/kit_ty_kate/.opam/default/lib/base/base_internalhash_types -I /Users/kit_ty_kate/.opam/default/lib/base/caml -I /Users/kit_ty_kate/.opam/default/lib/base/md5 -I /Users/kit_ty_kate/.opam/default/lib/base/shadow_stdlib -I /Users/kit_ty_kate/.opam/default/lib/base_bigstring -I /Users/kit_ty_kate/.opam/default/lib/base_quickcheck -I /Users/kit_ty_kate/.opam/default/lib/base_quickcheck/ppx_quickcheck/runtime -I /Users/kit_ty_kate/.opam/default/lib/bin_prot -I /Users/kit_ty_kate/.opam/default/lib/bin_prot/shape -I /Users/kit_ty_kate/.opam/default/lib/core -I /Users/kit_ty_kate/.opam/default/lib/core/base_for_tests -I /Users/kit_ty_kate/.opam/default/lib/core/validate -I /Users/kit_ty_kate/.opam/default/lib/core_kernel/caml_unix -I /Users/kit_ty_kate/.opam/default/lib/fieldslib -I /Users/kit_ty_kate/.opam/default/lib/int_repr -I /Users/kit_ty_kate/.opam/default/lib/jane-street-headers -I /Users/kit_ty_kate/.opam/default/lib/parsexp -I /Users/kit_ty_kate/.opam/default/lib/ppx_assert/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_bench/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_compare/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_enumerate/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_expect/collector -I /Users/kit_ty_kate/.opam/default/lib/ppx_expect/common -I /Users/kit_ty_kate/.opam/default/lib/ppx_expect/config -I /Users/kit_ty_kate/.opam/default/lib/ppx_expect/config_types -I /Users/kit_ty_kate/.opam/default/lib/ppx_hash/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_here/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_inline_test/config -I /Users/kit_ty_kate/.opam/default/lib/ppx_inline_test/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_log/types -I /Users/kit_ty_kate/.opam/default/lib/ppx_module_timer/runtime -I /Users/kit_ty_kate/.opam/default/lib/ppx_sexp_conv/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/sexplib -I /Users/kit_ty_kate/.opam/default/lib/sexplib0 -I /Users/kit_ty_kate/.opam/default/lib/splittable_random -I /Users/kit_ty_kate/.opam/default/lib/stdio -I /Users/kit_ty_kate/.opam/default/lib/time_now -I /Users/kit_ty_kate/.opam/default/lib/typerep -I /Users/kit_ty_kate/.opam/default/lib/variantslib -I ../../sys_unix/src -o filename_unix_stubs.o -c filename_unix_stubs.c)
filename_unix_stubs.c:32:9: warning: initializing 'char *' with an expression of type 'const char *' discards qualifiers [-Wincompatible-pointer-types-discards-qualifiers]
  char *path = String_val(v_path);
        ^      ~~~~~~~~~~~~~~~~~~
1 warning generated.
File "core_unix/src/dune", line 5, characters 45-65:
5 |  (c_names nss_stubs timespec core_unix_stubs core_unix_time_stubs)
                                                 ^^^^^^^^^^^^^^^^^^^^
(cd _build/default/core_unix/src && /usr/bin/cc -O2 -fno-strict-aliasing -fwrapv -pthread -D_FILE_OFFSET_BITS=64 -O2 -fno-strict-aliasing -fwrapv -pthread -D_LARGEFILE64_SOURCE -g -I /Users/kit_ty_kate/.opam/default/lib/ocaml -I /Users/kit_ty_kate/.opam/default/lib/base -I /Users/kit_ty_kate/.opam/default/lib/base/base_internalhash_types -I /Users/kit_ty_kate/.opam/default/lib/base/caml -I /Users/kit_ty_kate/.opam/default/lib/base/md5 -I /Users/kit_ty_kate/.opam/default/lib/base/shadow_stdlib -I /Users/kit_ty_kate/.opam/default/lib/base_bigstring -I /Users/kit_ty_kate/.opam/default/lib/base_quickcheck -I /Users/kit_ty_kate/.opam/default/lib/base_quickcheck/ppx_quickcheck/runtime -I /Users/kit_ty_kate/.opam/default/lib/bin_prot -I /Users/kit_ty_kate/.opam/default/lib/bin_prot/shape -I /Users/kit_ty_kate/.opam/default/lib/core -I /Users/kit_ty_kate/.opam/default/lib/core/base_for_tests -I /Users/kit_ty_kate/.opam/default/lib/core/validate -I /Users/kit_ty_kate/.opam/default/lib/core_kernel/caml_threads -I /Users/kit_ty_kate/.opam/default/lib/core_kernel/caml_unix -I /Users/kit_ty_kate/.opam/default/lib/core_kernel/flags -I /Users/kit_ty_kate/.opam/default/lib/fieldslib -I /Users/kit_ty_kate/.opam/default/lib/int_repr -I /Users/kit_ty_kate/.opam/default/lib/jane-street-headers -I /Users/kit_ty_kate/.opam/default/lib/ocaml/threads -I /Users/kit_ty_kate/.opam/default/lib/parsexp -I /Users/kit_ty_kate/.opam/default/lib/ppx_assert/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_bench/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_compare/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_enumerate/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_expect/collector -I /Users/kit_ty_kate/.opam/default/lib/ppx_expect/common -I /Users/kit_ty_kate/.opam/default/lib/ppx_expect/config -I /Users/kit_ty_kate/.opam/default/lib/ppx_expect/config_types -I /Users/kit_ty_kate/.opam/default/lib/ppx_hash/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_here/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_inline_test/config -I /Users/kit_ty_kate/.opam/default/lib/ppx_inline_test/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_log/types -I /Users/kit_ty_kate/.opam/default/lib/ppx_module_timer/runtime -I /Users/kit_ty_kate/.opam/default/lib/ppx_sexp_conv/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/sexplib -I /Users/kit_ty_kate/.opam/default/lib/sexplib/unix -I /Users/kit_ty_kate/.opam/default/lib/sexplib0 -I /Users/kit_ty_kate/.opam/default/lib/spawn -I /Users/kit_ty_kate/.opam/default/lib/splittable_random -I /Users/kit_ty_kate/.opam/default/lib/stdio -I /Users/kit_ty_kate/.opam/default/lib/time_now -I /Users/kit_ty_kate/.opam/default/lib/typerep -I /Users/kit_ty_kate/.opam/default/lib/variantslib -I ../../error_checking_mutex/src -I ../../signal_unix/src -o core_unix_time_stubs.o -c core_unix_time_stubs.c)
core_unix_time_stubs.c:57:13: error: implicit declaration of function 'clock_getcpuclockid' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
  int ret = clock_getcpuclockid(pid, &clock);
            ^
core_unix_time_stubs.c:57:13: note: did you mean 'caml_clock_getcpuclockid'?
core_unix_time_stubs.c:52:16: note: 'caml_clock_getcpuclockid' declared here
CAMLprim value caml_clock_getcpuclockid(value v_pid) {
               ^
1 error generated.
(cd _build/default && /Users/kit_ty_kate/.opam/default/bin/ocamlmklib.opt -g -o core_thread/src/core_thread_stubs core_thread/src/pthread_np_stubs.o)
warning: /Library/Developer/CommandLineTools/usr/bin/ranlib: archive library: core_thread/src/libcore_thread_stubs.a the table of contents is empty (no object file members in the library define global symbols)
warning: /Library/Developer/CommandLineTools/usr/bin/ranlib: archive library: core_thread/src/libcore_thread_stubs.a the table of contents is empty (no object file members in the library define global symbols)
(cd _build/default && /Users/kit_ty_kate/.opam/default/bin/ocamlmklib.opt -g -o linux_ext/src/linux_ext_stubs linux_ext/src/linux_ext_stubs.o)
warning: /Library/Developer/CommandLineTools/usr/bin/ranlib: archive library: linux_ext/src/liblinux_ext_stubs.a the table of contents is empty (no object file members in the library define global symbols)
warning: /Library/Developer/CommandLineTools/usr/bin/ranlib: archive library: linux_ext/src/liblinux_ext_stubs.a the table of contents is empty (no object file members in the library define global symbols)
(cd _build/default/core_unix/src && /usr/bin/cc -O2 -fno-strict-aliasing -fwrapv -pthread -D_FILE_OFFSET_BITS=64 -O2 -fno-strict-aliasing -fwrapv -pthread -D_LARGEFILE64_SOURCE -g -I /Users/kit_ty_kate/.opam/default/lib/ocaml -I /Users/kit_ty_kate/.opam/default/lib/base -I /Users/kit_ty_kate/.opam/default/lib/base/base_internalhash_types -I /Users/kit_ty_kate/.opam/default/lib/base/caml -I /Users/kit_ty_kate/.opam/default/lib/base/md5 -I /Users/kit_ty_kate/.opam/default/lib/base/shadow_stdlib -I /Users/kit_ty_kate/.opam/default/lib/base_bigstring -I /Users/kit_ty_kate/.opam/default/lib/base_quickcheck -I /Users/kit_ty_kate/.opam/default/lib/base_quickcheck/ppx_quickcheck/runtime -I /Users/kit_ty_kate/.opam/default/lib/bin_prot -I /Users/kit_ty_kate/.opam/default/lib/bin_prot/shape -I /Users/kit_ty_kate/.opam/default/lib/core -I /Users/kit_ty_kate/.opam/default/lib/core/base_for_tests -I /Users/kit_ty_kate/.opam/default/lib/core/validate -I /Users/kit_ty_kate/.opam/default/lib/core_kernel/caml_threads -I /Users/kit_ty_kate/.opam/default/lib/core_kernel/caml_unix -I /Users/kit_ty_kate/.opam/default/lib/core_kernel/flags -I /Users/kit_ty_kate/.opam/default/lib/fieldslib -I /Users/kit_ty_kate/.opam/default/lib/int_repr -I /Users/kit_ty_kate/.opam/default/lib/jane-street-headers -I /Users/kit_ty_kate/.opam/default/lib/ocaml/threads -I /Users/kit_ty_kate/.opam/default/lib/parsexp -I /Users/kit_ty_kate/.opam/default/lib/ppx_assert/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_bench/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_compare/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_enumerate/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_expect/collector -I /Users/kit_ty_kate/.opam/default/lib/ppx_expect/common -I /Users/kit_ty_kate/.opam/default/lib/ppx_expect/config -I /Users/kit_ty_kate/.opam/default/lib/ppx_expect/config_types -I /Users/kit_ty_kate/.opam/default/lib/ppx_hash/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_here/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_inline_test/config -I /Users/kit_ty_kate/.opam/default/lib/ppx_inline_test/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/ppx_log/types -I /Users/kit_ty_kate/.opam/default/lib/ppx_module_timer/runtime -I /Users/kit_ty_kate/.opam/default/lib/ppx_sexp_conv/runtime-lib -I /Users/kit_ty_kate/.opam/default/lib/sexplib -I /Users/kit_ty_kate/.opam/default/lib/sexplib/unix -I /Users/kit_ty_kate/.opam/default/lib/sexplib0 -I /Users/kit_ty_kate/.opam/default/lib/spawn -I /Users/kit_ty_kate/.opam/default/lib/splittable_random -I /Users/kit_ty_kate/.opam/default/lib/stdio -I /Users/kit_ty_kate/.opam/default/lib/time_now -I /Users/kit_ty_kate/.opam/default/lib/typerep -I /Users/kit_ty_kate/.opam/default/lib/variantslib -I ../../error_checking_mutex/src -I ../../signal_unix/src -o core_unix_stubs.o -c core_unix_stubs.c)
core_unix_stubs.c:349:2: warning: "_POSIX_SYNCHRONIZED_IO undefined or <= 0; aliasing unix_fdatasync to unix_fsync" [-W#warnings]
#warning "_POSIX_SYNCHRONIZED_IO undefined or <= 0; aliasing unix_fdatasync to unix_fsync"
 ^
core_unix_stubs.c:972:49: warning: passing 'gid_t [16]' to parameter of type 'int *' converts between pointers to integer types with different sign [-Wpointer-sign]
    n = getgrouplist(c_user, Long_val(v_group), groups, &ngroups);
                                                ^~~~~~
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/unistd.h:654:43: note: passing argument to parameter here
int      getgrouplist(const char *, int, int *, int *);
                                              ^
core_unix_stubs.c:1282:2: warning: "_POSIX_PRIORITY_SCHEDULING not present; sched_setscheduler undefined" [-W#warnings]
#warning "_POSIX_PRIORITY_SCHEDULING not present; sched_setscheduler undefined"
 ^
core_unix_stubs.c:1285:4: warning: "invalid_argument" is deprecated: use "caml_invalid_argument" instead [-W#pragma-messages]
{  invalid_argument("sched_setscheduler unimplemented"); }
   ^
/Users/kit_ty_kate/.opam/default/lib/ocaml/caml/compatibility.h:112:26: note: expanded from macro 'invalid_argument'
#define invalid_argument CAML_DEPRECATED("invalid_argument", "caml_invalid_argument") caml_invalid_argument
                         ^
/Users/kit_ty_kate/.opam/default/lib/ocaml/caml/misc.h:57:3: note: expanded from macro 'CAML_DEPRECATED'
  CAML_PREPROWARNING(name1 is deprecated: use name2 instead)
  ^
/Users/kit_ty_kate/.opam/default/lib/ocaml/caml/misc.h:55:31: note: expanded from macro 'CAML_PREPROWARNING'
#define CAML_PREPROWARNING(x) _Pragma(CAML_MAKEWARNING2(x))
                              ^
<scratch space>:59:6: note: expanded from here
 GCC warning "\"invalid_argument\" is deprecated: use \"caml_invalid_argument\" instead"
     ^
4 warnings generated.
(cd _build/default/unix_pseudo_terminal/src && /bin/bash -e -u -o pipefail -c './discover.sh config_ext.h /Users/kit_ty_kate/.opam/default/lib/jst-config/config.h')
getconf: no such configuration parameter `GNU_LIBC_VERSION'
```

File: packages/ocaml_intrinsics/ocaml_intrinsics.v0.15.0/opam (hunk 1 above)
Comment (gretay-js) [reply packages/ocaml_intrinsics/ocaml_intrinsics.v0.15.0/opam:17]: > this also constraints newer macs out (arm64)

indeed, this fix doesn't address the problem on the new macos on arm targets.

File: packages/ocaml_intrinsics/ocaml_intrinsics.v0.15.1/opam [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+license: "MIT"
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+]
+depends: [
+  "ocaml"             {>= "4.08.0"}
+  "dune"              {>= "2.0.0"}
+  "dune-configurator"
+]
+available: (arch = "x86_32" | arch = "x86_64") & os != "win32"
```
Comment (kit-ty-kate) [inline packages/ocaml_intrinsics/ocaml_intrinsics.v0.15.1/opam:17]: Does it also work on macOS/arm64?
```suggestion
available: (arch = "x86_32" | arch = "x86_64" | arch = "arm64") & os != "win32"
```

Comment (gretay-js) [reply packages/ocaml_intrinsics/ocaml_intrinsics.v0.15.1/opam:17]: I haven't been able to test it yet on amcOS/arm64, I don't have access to one. Can we fix it for x86 first, or do you prefer to wait until arm64 is fixed and tested?

**Comment by kit-ty-kate** [comment]
marking as "do not merge" until we figure out failures in core_unix happening on both macOS/x86_64 and arm64

**Comment by gretay-js** [comment]
> marking as "do not merge" until we figure out failures in core_unix happening on both macOS/x86_64 and arm64

@kit-ty-kate  The failures in `core_unix` are unrelated to intrinsics. Can we move the constraint `os != win32` to `core_unix` and merge this PR?

**Comment by kit-ty-kate** [comment]
sure

**Comment by gretay-js** [comment]
> sure

great, thanks! Also, it should be fine to enable intrinsics on macOS/M1. I'll update this PR for both.

File: packages/ocaml_intrinsics/ocaml_intrinsics.v0.15.1/opam (hunk 2 above)
Comment (kit-ty-kate) [inline packages/ocaml_intrinsics/ocaml_intrinsics.v0.15.1/opam:17]: We can let CI see if it works or not
```suggestion
available: os != "win32"
```

**Comment by kit-ty-kate** [comment]
Thanks

File: packages/ocaml_intrinsics/ocaml_intrinsics.v0.15.1/opam (hunk 2 above)
Comment (anuragsoni) [reply packages/ocaml_intrinsics/ocaml_intrinsics.v0.15.1/opam:17]: If it helps, i've been using `ocaml_intrinsics` on an arm64 mac (m1) with the patch version https://github.com/janestreet/ocaml_intrinsics/commit/7d7668a753c1ada0d1d79e7e1fa6a3eddb61d349

I haven't noticed any issues and the tests from https://github.com/yallop/intrinsics-report pass on the m1 mac.

**Pull request merged**